modules/internal/CString.chpl

```diff
@@ -120,13 +120,29 @@ module CString {
     return __primitive("cast", t, x);
   }
   //
+  // casts from c_string to c_ptrConst(c_char/int(8)/uint(8))
+  //
+  inline operator :(x: c_string, type t:c_ptrConst)
+    where t.eltType == c_char || t.eltType == int(8) || t.eltType == uint(8)
```

Comment:
Not requesting a change, but I think you can do `type t: c_ptrConst(?eltType)` and use `eltType` in the `where` clause.

Reply:
I will go ahead and change that in this PR, here and for the corresponding `c_ptr` versions